client/src/components/Page/PageDropdown.vue

```diff
@@ -0,0 +1,85 @@
+<script setup lang="ts">
+import _l from "@/utils/localization";
+import { computed, unref } from "vue";
+import { deletePage } from "./services";
+import { storeToRefs } from "pinia";
+import { useUserStore } from "@/stores/userStore";
+import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
+import { faCaretDown } from "@fortawesome/free-solid-svg-icons";
+import { library } from "@fortawesome/fontawesome-svg-core";
+
+library.add(faCaretDown);
+
+interface Page {
+    id: string;
+    shared?: Boolean;
+    title?: string;
+    description?: string;
+}
+interface PageDropdownProps {
+    page: Page;
+    root: string;
+    published?: Boolean;
+}
+
+const props = defineProps<PageDropdownProps>();
+const { isAnonymous } = storeToRefs(useUserStore());
+
+const emit = defineEmits(["onRemove", "onSuccess", "onError"]);
+
+const urlEdit = computed(() => `${props.root}pages/editor?id=${props.page.id}`);
+const urlEditAttributes = computed(() => `${props.root}pages/edit?id=${props.page.id}`);
+const urlShare = computed(() => `${props.root}pages/sharing?id=${props.page.id}`);
+const urlView = computed(() => `${props.root}published/page?id=${props.page.id}`);
+const readOnly = computed(() => props.page.shared || props.published || unref(isAnonymous));
+
+function onDelete() {
+    const confirmationMessage = _l(`Are you sure you want to delete page`) + ` "${props.page.title}"?`;
+    if (window.confirm(confirmationMessage)) {
```

Comment:
I think it would be better to utilize the modal instead of using the browser confirmation alert for deletion.

Reply:
@itisAliRH Do you mean a specific modal usage? Like our ui-modal.js component? I generally prefer the default for such simple use cases because it has well-defined and user-known display and shortcuts in all browsers.

Reply:
@martenson I mean using bootstrap modal; e.g. [here](https://github.com/galaxyproject/galaxy/blob/dev/client/src/components/History/CurrentHistory/HistoryNavigation.vue#L161) it's a simple use case too but using a single, unified modal component is beneficial because it promotes consistency across the application.

Reply:
I like that, thanks for the suggestion!